modules/js/generator/CMakeLists.txt

```diff
@@ -38,8 +38,23 @@ set(scripts_hdr_parser "${JS_SOURCE_DIR}/../python/src2/hdr_parser.py")
 
 if(DEFINED ENV{OPENCV_JS_WHITELIST})
   set(OPENCV_JS_WHITELIST_FILE "$ENV{OPENCV_JS_WHITELIST}")
+  message(STATUS "Use white list from environment ${OPENCV_JS_WHITELIST_FILE}")
 else()
-  set(OPENCV_JS_WHITELIST_FILE "${OpenCV_SOURCE_DIR}/platforms/js/opencv_js.config.py")
+  #generate white list from modules/<module_name>/misc/js/whitelist.json
+  set(OPENCV_JS_WHITELIST_FILE "${CMAKE_CURRENT_BINARY_DIR}/whitelist.json")
+  set(OPENCV_JS_WHITELIST_CONTENT)
+  file(WRITE "${OPENCV_JS_WHITELIST_FILE}" "{\n")
```

Comment:
No need to write file each time.
Generate string and update on changes only. See `ocv_update_file`

Also generated file should be added as a dependency to the build graph of bindings generator target.